src/runtime/program/vote/execute.zig

```diff
@@ -138,10 +174,287 @@ fn intializeAccount(
+    current_epoch: Epoch,
+) InstructionError!void {
+    const authorized_withdrawer_signer = if (signers) |signers_|
+        try verifyAuthorizedSigner(vote_state.authorized_withdrawer, signers_)
+    else
+        ic.info.isPubkeySigner(vote_state.authorized_withdrawer);
+
+    // current authorized withdrawer or epoch authorized voter must say "yay"
+    if (!authorized_withdrawer_signer) {
+        const epoch_authorized_voter = try vote_state.getAndUpdateAuthorizedVoter(
+            allocator,
+            current_epoch,
+        );
+        if (!ic.info.isPubkeySigner(epoch_authorized_voter)) {
+            return InstructionError.MissingRequiredSignature;
+        }
+    }
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L82-L92
+fn executeAuthorizeWithSeed(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    new_account: Pubkey,
+    authorization_type: VoteAuthorize,
+    current_authority_derived_key_owner: Pubkey,
+    current_authority_derived_key_seed: []const u8,
+) InstructionError!void {
+    try ic.info.checkNumberOfAccounts(3);
+
+    try authorizeWithSeed(
+        allocator,
+        ic,
+        vote_account,
+        new_account,
+        authorization_type,
+        current_authority_derived_key_owner,
+        current_authority_derived_key_seed,
+        @intFromEnum(vote_instruction.VoteAuthorizeWithSeedArgs.AccountIndex.signer),
+        @intFromEnum(vote_instruction.VoteAuthorizeWithSeedArgs.AccountIndex.clock_sysvar),
+    );
+}
+
+/// Analogous to [process_authorize_with_seed_instruction] https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L19
+fn authorizeWithSeed(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    new_authority: Pubkey,
+    authorization_type: VoteAuthorize,
+    owner: Pubkey,
+    seed: []const u8,
+    signer_index: u8,
+    clock_index: u8,
+) InstructionError!void {
+    const clock = try ic.getSysvarWithAccountCheck(Clock, clock_index);
+
+    var expected_authority_keys = std.AutoHashMap(Pubkey, void).init(allocator);
+    defer expected_authority_keys.deinit();
+    if (try ic.info.isIndexSigner(signer_index)) {
+        var signer_account = try ic.borrowInstructionAccount(signer_index);
+        defer signer_account.release();
+        const created = pubkey_utils.createWithSeed(
+            signer_account.pubkey,
+            seed,
+            owner,
+        ) catch |err| {
+            ic.tc.custom_error = @intFromError(err);
+            return InstructionError.Custom;
+        };
+        expected_authority_keys.put(created, {}) catch |err| {
+            // TODO okay to convert out of memory to custom error?
+            ic.tc.custom_error = @intFromError(err);
+            return InstructionError.Custom;
+        };
```

Comment:
```suggestion
        var signer_account = try ic.borrowInstructionAccount(signer_index);
        defer signer_account.release();
        
        const created = pubkey_utils.createWithSeed(
            signer_account.pubkey,
            seed,
            owner,
        ) catch |err| {
            ic.tc.custom_error = @intFromError(err);
            return InstructionError.Custom;
        };
        
        expected_authority_keys.put(created, {}) catch |err| {
            // TODO okay to convert out of memory to custom error?
            ic.tc.custom_error = @intFromError(err);
            return InstructionError.Custom;
        };
```

small nit on the spacing

Reply:
[c711769](https://github.com/Syndica/sig/pull/576/commits/c71176924ce327a39bfa331a1129ca68fe7473c8)